third_party/purge/mcp.purge.php

```diff
@@ -0,0 +1,110 @@
+<?php  if ( ! defined('BASEPATH')) exit('No direct script access allowed');
+
+
+ 
+class Purge_mcp {
+	
+	public $return_data;
+	public $return_array = array();
+	
+	private $_base_url;
+	private $_data = array();
+	private $_module = 'purge';
+	
+	
+	/**
+	 * Constructor
+	 */
+	public function __construct()
+	{
+		$this->EE =& get_instance();
+		
+		$this->_base_url = BASE.AMP.'C=addons_modules'.AMP.'M=show_module_cp'.AMP.'module=purge';
+		
+		$this->EE->cp->set_right_nav(array(
+			'module_home'	=> $this->_cp_url()
+		));
+
+		$this->EE->view->cp_page_title = "Purge: Channel URL Patterns for Varnish";
+	}
+	
+	
+
+
+	public function index()
+	{
+		
+		$this->EE->load->library('table');
+		
+		$this->_data['action_url'] = $this->_base_url . '&method=save';
+		//$this->_data['images'] = $this->get_alttags();
+		
+		$this->EE->load->model('channel_model');
+		$channels_query = $this->EE->channel_model->get_channels()->result();
+		foreach ($channels_query as $channel) 
+			$this->_data['channels'][] = array('channel_title' => $channel->channel_title, 'channel_name' => $channel->channel_name, 'channel_id' => $channel->channel_id );
+		
+		$this->_data['rules'] = $this->get();
+		
+		return $this->EE->load->view('rules', $this->_data, TRUE);
+		
+	}
+	
+	private function get()
+	{
+		$this->EE->db->select('*');
+		return $this->EE->db->get_where('purge_rules')->result_array();
+	}
+	
+	
+	public function save()
+	{	
```

Comment:
I'm not sure how I'd do this. The patterns will contain all url friendly characters, curly brackets and even regex.